test/src/test/java/hudson/tasks/LogRotatorTest.java

```diff
@@ -96,6 +101,17 @@ public void successVsFailureWithRemoveLastBuild() throws Exception {
         assertEquals(2, numberOf(project.getLastFailedBuild()));
     }
 
+    @Test
+    public void ableToDeleteCurrentBuild() throws Exception {
+        var p = j.createFreeStyleProject();
+        // Keep 0 builds, i.e. immediately delete builds as they complete.
+        LogRotator logRotator = new LogRotator(-1, 0, -1, -1);
+        logRotator.setRemoveLastBuild(true);
+        p.setBuildDiscarder(logRotator);
+        j.buildAndAssertStatus(Result.SUCCESS, p);
+        assertNull(p.getBuildByNumber(1));
```

Comment:
Failure in https://github.com/jenkinsci/jenkins/pull/9921/checks?check_run_id=32166988156 (#9921) looks like a flake? Missing `await`?

Reply:
I don't know what we would `await` for, because log rotation runs synchronously during build completion. Unless this is a test-specific timing issue due to `JenkinsRule.buildAndAssertSuccess` just waiting for `QueueTaskFuture.get` instead of `!Run.isLogUpdated`, then this test probably just needs to be skipped on Windows.

Reply:
From some testing, `j.buildAndAssertStatus(Result.SUCCESS, p)` is guaranteed to block until `GlobalBuildDiscarderListener.onFinalized` has completed. Perhaps the `BuildWatcher` background thread tried to copy the logs at an inopportune time and caused log rotation to fail. Seems best to delete the `BuildWatcher` here in case it made any of the other tests flaky on Windows and then probably skip this test explicitly on Windows as well.

Reply:
Also the `AccessDeniedException` seems strange. Either way, I filed https://github.com/jenkinsci/jenkins/pull/9923.